src/Psalm/Internal/LanguageServer/LanguageClient.php

```diff
@@ -16,11 +17,48 @@ class LanguageClient
      */
     public $textDocument;
 
+    /**
+     * The client handler
+     *
+     * @var ClientHandler
+     */
+    private $handler;
+
     public function __construct(ProtocolReader $reader, ProtocolWriter $writer)
     {
-        $handler = new ClientHandler($reader, $writer);
+        $this->handler = new ClientHandler($reader, $writer);
         $mapper = new JsonMapper;
 
-        $this->textDocument = new Client\TextDocument($handler, $mapper);
+        $this->textDocument = new Client\TextDocument($this->handler, $mapper);
+    }
+
+    /**
+     * Send a log message to the client.
+     *
+     * @param string $message The message to send to the client.
+     * @psalm-param 1|2|3|4 $type
+     * @param integer $type The log type:
+     *  - 1 = Error
+     *  - 2 = Warning
+     *  - 3 = Info
+     *  - 4 = Log
+
+     * @return Promise<void>
+     */
+    public function logMessage(string $message, int $type = 4, string $method = 'window/logMessage'): Promise
```

Comment:
This method allows us to send notifications with the LogMessageParams type as the params for the message.  The method is customizable, because we use this same method to send telemetry data using the `telemetry/event` method which can use anything for its params.